### PR TITLE
DBZ-3161 Replace MonitoredTables metrics with CapturedTables metrics

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-3161

### CODE REVIEW

**Comment by jpechane** [comment]
@ani-sha Looks good, could you please split it into more commits? One will contain the change to `StreamingChangeEventSourceMetrics.java` another adding the new dtabase history option and another one the refactoring/renaming of the parts of the code?

Also is still `store.only.monitored.tables.ddl` supported? It should be read and emit warning when set as we do with whitelist/blacklist settings.

And last but not least the documentation needs to be update with renamed metrics and history config option.

**Comment by ani-sha** [comment]
Sure, @jpechane 
> Also is still store.only.monitored.tables.ddl supported?

yes, it's will be still supported, I will add the warnings. 

> documentation needs to be update with renamed metrics and history config option.

As the option is deprecated, should it remain in the docs and I will just add details about the new option. Or should the old option be renamed to the new option?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/BinlogReaderMetrics.java [hunk 1]
```diff
@@ -152,8 +152,14 @@ public void setMilliSecondsBehindSource(long value) {
     }
 
     @Override
+    @Deprecated
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/BinlogReaderMetrics.java:155]: Could you add a JavaDoc tag referring to the replacement, e.g.:

    @deprecated("Superseded by the 'Captured Tables' metric. Scheduled for removal in a future release")

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/ReaderMetricsMXBean.java [hunk 2]
```diff
@@ -13,5 +13,9 @@
  */
 public interface ReaderMetricsMXBean {
 
+    @Deprecated
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/ReaderMetricsMXBean.java:16]: Same comment as above.

File: debezium-core/src/main/java/io/debezium/pipeline/metrics/ChangeEventSourceMetricsMXBean.java [hunk 3]
```diff
@@ -22,8 +22,11 @@
 
     long getNumberOfErroneousEvents();
 
+    @Deprecated
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/metrics/ChangeEventSourceMetricsMXBean.java:25]: Same comment as above.

File: debezium-core/src/main/java/io/debezium/pipeline/metrics/SnapshotChangeEventSourceMetrics.java [hunk 4]
```diff
@@ -85,8 +85,14 @@ public long getSnapshotDurationInSeconds() {
     }
 
     @Override
+    @Deprecated
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/metrics/SnapshotChangeEventSourceMetrics.java:88]: same comment as above.

File: debezium-core/src/main/java/io/debezium/pipeline/metrics/StreamingChangeEventSourceMetrics.java [hunk 5]
```diff
@@ -51,10 +51,16 @@ public boolean isConnected() {
     }
 
     @Override
+    @Deprecated
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/metrics/StreamingChangeEventSourceMetrics.java:54]: Same comment as above.

File: debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistory.java [hunk 6]
```diff
@@ -146,7 +158,17 @@
      */
     void initializeStorage();
 
-    boolean storeOnlyMonitoredTables();
+    boolean storeOnlyCapturedTables();
 
     boolean skipUnparseableDdlStatements();
+
+    static int validateMonitoredTables(Configuration config, Field field, Field.ValidationOutput problems) {
+        String msg = config.getFallbackStringPropertyWithWarning(STORE_ONLY_CAPTURED_TABLES_DDL, STORE_ONLY_MONITORED_TABLES_DDL);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistory.java:166]: Won't this raise the warning too if only STORE_ONLY_CAPTURED_TABLES_DDL has been given? I think you'd want to only query for STORE_ONLY_MONITORED_TABLES_DDL, without any fallback?

**Review by gunnarmorling — COMMENTED** [review]
LGTM overall, @ani-sha! Some small comments inline. Re your question on docs, let's keep the old one for now there, adding the new one, and marking the old one as deprecated, using a sentence similar to the one I suggested for JavaDoc.

File: debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistory.java (hunk 6 above)
Comment (ani-sha) [reply debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistory.java:166]: The warning will be only emitted when both/only `STORE_ONLY_MONITORED_TABLES_DDL` is provided as this validation check is added to the deprecated field only.

File: debezium-core/src/main/java/io/debezium/relational/history/AbstractDatabaseHistory.java [hunk 7]
```diff
@@ -64,7 +64,7 @@ public void configure(Configuration config, HistoryRecordComparator comparator,
         this.config = config;
         this.comparator = comparator != null ? comparator : HistoryRecordComparator.INSTANCE;
         this.skipUnparseableDDL = config.getBoolean(DatabaseHistory.SKIP_UNPARSEABLE_DDL_STATEMENTS);
-        this.storeOnlyMonitoredTablesDdl = config.getBoolean(DatabaseHistory.STORE_ONLY_MONITORED_TABLES_DDL);
+        this.storeOnlyCapturedTablesDdl = config.getBoolean(DatabaseHistory.STORE_ONLY_CAPTURED_TABLES_DDL);
```
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/relational/history/AbstractDatabaseHistory.java]: I'll push a change for this and some other minor adjustments.

**Review by gunnarmorling — COMMENTED** [review]
Pushed one more commit with a few changes.

**Comment by gunnarmorling** [comment]
@jpechane, I've done some updates. Could you double check and merge if it's good? Thx!

**Comment by jpechane** [comment]
@ani-sha @gunnarmorling Applied, thanks!

**Pull request merged**